engine/utils/utils.hpp

```diff
@@ -153,7 +153,15 @@ inline std::string string_view_2_string(const StringView& src) {
 inline int compare_string_view(const StringView& src,
                                const StringView& target) {
   auto size = std::min(src.size(), target.size());
-  for (uint32_t i = 0; i < size; i++) {
+  auto batchNum = size & ~(sizeof(uint64_t)-1);
+  uint32_t i = 0;
+  for( i =0; i< batchNum ; ){
+    auto nEq = *(uint64_t*)(&src[i]) - *(uint64_t*)(&target[i]);
+    if (nEq)
```

Comment:
The algorithm is incorrect.
x86 systems adopts little endianness. In other words, you have to reverse 8 bytes first, then treat them as a uint64_t for comparison.
Also, when uint64_t converts to int, overflow may occur and the result will be wrong. For example, `"Hello, World!"` and `"Hello, world!"` will be equal under this algorithm.

Reply:
OK. I thought it was only used for equivalence compare.
On little endian host the return value is the opposite.